### PR TITLE
Prevent "source changed" message for every version change when using trunk

### LINKED ISSUE #9865: Source changed message when source didn't change
🌈
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

`pod install`
## What did you expect to happen?

```
Installing Realm 5.0.3 (was 5.0.2)
```

## What happened instead?

```
Installing Realm 5.0.3 (was 5.0.2 and source changed to `https://cdn.cocoapods.org/` from `trunk`)
```

### Stack

```
   CocoaPods : 1.9.3
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.4 (19E287)
       Xcode : 11.5 (11E608c)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /usr/local/var/rbenv/versions/2.6.3/lib
Repositories : trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/var/rbenv/versions/2.6.3/bin/pod
```

### Plugins

```
cocoapods-deintegrate  : 1.0.4
cocoapods-dependencies : 1.3.0
cocoapods-plugins      : 1.0.0
cocoapods-search       : 1.0.0
cocoapods-stats        : 1.1.0
cocoapods-trunk        : 1.5.0
cocoapods-try          : 1.2.0
slather                : 2.4.8
```

### Podfile

```ruby
source 'https://cdn.cocoapods.org/'

platform :ios, "13.1"
use_frameworks! :linkage => :static

target 'TargetName' do
  pod 'RealmSwift', "5.0.3"
end
```

### PR DESCRIPTION
**Pull request by cltnschlosser**
Resolves #9865 

Based off the logic in cocoapods-core for creating lockfiles.
```ruby
      def generate_spec_repos(spec_repos)
        Hash[spec_repos.map do |source, specs|
          next unless source
          next if specs.empty?
          key = source.url || source.name

          # save `trunk` as 'trunk' so that the URL itself can be changed without lockfile churn
          key = Pod::TrunkSource::TRUNK_REPO_NAME if source.name == Pod::TrunkSource::TRUNK_REPO_NAME

          value = specs.map { |s| s.root.name }.uniq
          [key, YAMLHelper.sorted_array(value)]
        end.compact]
      end
```

### CODE REVIEW

**Comment by cltnschlosser** [comment]
cc @dnkoutso

File: lib/cocoapods/installer.rb [hunk 1]
```diff
@@ -498,6 +498,7 @@ def install_pod_sources
             previous_version = sandbox.manifest.version(spec.name)
             has_changed_version = current_version != previous_version
             current_repo = analysis_result.specs_by_source.detect { |key, values| break key if values.map(&:name).include?(spec.name) }
+            current_repo &&= Pod::TrunkSource::TRUNK_REPO_NAME if current_repo.name == Pod::TrunkSource::TRUNK_REPO_NAME
             current_repo &&= current_repo.url || current_repo.name
```
Comment (cltnschlosser) [inline lib/cocoapods/installer.rb:502]: I need to do an if else here, but rubocop won't let me. How is this suppose to be done?

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Prevent "source changed" message for every version change when using trunk  
  [cltnschlosser](https://github.com/cltnschlosser)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer.rb (hunk 1 above)
Comment (cltnschlosser) [reply lib/cocoapods/installer.rb]: I guess onelining was the solution

**Comment by dnkoutso** [comment]
Thanks!

**Pull request merged**